pkg/client/liquidation_recovery_test.go

```diff
@@ -0,0 +1,506 @@
+			configureBitcoinHandle: func() *localBitcoinConnection {
+				return newLocalBitcoinConnection()
+			},
+		},
+		"bitcoin extended public keys and working bitcoin connection": {
+			bitcoinAddressesOrKeys: bitcoinExtenderPublicKeys,
+			configureBitcoinHandle: func() *localBitcoinConnection {
+				return newLocalBitcoinConnection()
+			},
+		},
+		// bitcoin connection not working: failing IsAddressUnused
+		"bitcoin addresses and failing bitcoin call to IsAddressUnused": {
+			bitcoinAddressesOrKeys: bitcoinAddresses,
+			configureBitcoinHandle: func() *localBitcoinConnection {
+				bitcoinHandle := newLocalBitcoinConnection()
+				bitcoinHandle.isAddressUnusedError = fmt.Errorf("mocked failure")
+
+				return bitcoinHandle
+			},
+		},
+		"bitcoin extended public keys and failing bitcoin call to IsAddressUnused": {
+			bitcoinAddressesOrKeys: bitcoinExtenderPublicKeys,
+			configureBitcoinHandle: func() *localBitcoinConnection {
+				bitcoinHandle := newLocalBitcoinConnection()
+				bitcoinHandle.isAddressUnusedError = fmt.Errorf("mocked failure")
+
+				return bitcoinHandle
+			},
+		},
+		// bitcoin connection not working: failing VbyteFeeFor25Blocks
+		"bitcoin addresses and failing bitcoin call to VbyteFeeFor25Blocks": {
+			bitcoinAddressesOrKeys: bitcoinAddresses,
+			configureBitcoinHandle: func() *localBitcoinConnection {
+				bitcoinHandle := newLocalBitcoinConnection()
+				bitcoinHandle.vbyteFeeFor25BlocksError = fmt.Errorf("mocked failure")
+
+				return bitcoinHandle
+			},
+		},
+		"bitcoin extended public keys and failing bitcoin call to VbyteFeeFor25Blocks": {
+			bitcoinAddressesOrKeys: bitcoinExtenderPublicKeys,
+			configureBitcoinHandle: func() *localBitcoinConnection {
+				bitcoinHandle := newLocalBitcoinConnection()
+				bitcoinHandle.vbyteFeeFor25BlocksError = fmt.Errorf("mocked failure")
+
+				return bitcoinHandle
+			},
+		},
+		// bitcoin connection not working: failing Broadcast
+		"bitcoin addresses and failing bitcoin call to Broadcast": {
+			bitcoinAddressesOrKeys: bitcoinAddresses,
+			configureBitcoinHandle: func() *localBitcoinConnection {
+				bitcoinHandle := newLocalBitcoinConnection()
+				bitcoinHandle.broadcastError = fmt.Errorf("mocked failure")
+
+				return bitcoinHandle
+			},
+		},
+		"bitcoin extended public keys and failing bitcoin call to Broadcast": {
+			bitcoinAddressesOrKeys: bitcoinExtenderPublicKeys,
+			configureBitcoinHandle: func() *localBitcoinConnection {
+				bitcoinHandle := newLocalBitcoinConnection()
+				bitcoinHandle.broadcastError = fmt.Errorf("mocked failure")
+
+				return bitcoinHandle
+			},
+		},
+		// TODO: Add tests to verify logged output:
+		// - resolved maxFeePerVByte
+		// - logged 5x warn on broadcast failure
+		// - cover more failures
+	}
+
+	for testName, testData := range testCases {
+		t.Run(testName, func(t *testing.T) {
+			bitcoinHandle := testData.configureBitcoinHandle()
+
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
I think 5 secs might be pushing it for slower machines or when the tests are running under load

Reply:
It took 2 secs on my machine, but yeah 5 seconds may be risky. I'll bump it up to 10 secs maybe.